src/foam/nanos/crunch/Capability.js

```diff
@@ -356,24 +349,30 @@ foam.CLASS({
           subject.setUser((User) userDAO.find(ucj.getSourceId()));
         }
 
-        for ( CapabilityCapabilityJunction ccJunction : ccJunctions ) {
-          Capability cap = (Capability) ccJunction.findSourceId(x);
-          if ( ! cap.getEnabled() ) continue;
-          UserCapabilityJunction ucJunction = crunchService.getJunctionForSubject(x, ccJunction.getTargetId(), subject);
-
-          if ( ucJunction != null && ucJunction.getStatus() == CapabilityJunctionStatus.GRANTED )
-            continue;
-
-          if ( ucJunction == null ) {
-            return CapabilityJunctionStatus.ACTION_REQUIRED;
-          }
-          if ( ucJunction.getStatus() != CapabilityJunctionStatus.GRANTED
-               && ucJunction.getStatus() != CapabilityJunctionStatus.PENDING ) {
-            return CapabilityJunctionStatus.ACTION_REQUIRED;
+        var prereqs = crunchService.getPrereqs(getId());
+        if ( prereqs != null ) {
+          for ( var capId : prereqs ) {
+            var cap = (Capability) capabilityDAO.find(capId);
+            if ( cap == null || ! cap.getEnabled() ) continue;
+
+            UserCapabilityJunction ucJunction = crunchService.getJunctionForSubject(x, capId, subject);
+
+            if ( ucJunction != null && ucJunction.getStatus() == CapabilityJunctionStatus.GRANTED )
+              continue;
+
+            if ( ucJunction == null ) {
+              return CapabilityJunctionStatus.ACTION_REQUIRED;
+            }
+            if ( ucJunction.getStatus() != CapabilityJunctionStatus.GRANTED
+                 && ucJunction.getStatus() != CapabilityJunctionStatus.PENDING ) {
+              return CapabilityJunctionStatus.ACTION_REQUIRED;
+            }
+            if ( ucJunction.getStatus() == CapabilityJunctionStatus.PENDING ) {
```

Comment:
I think this should be kept the same, the idea is that if there is one or more prereqs in ACTION_REQUIRED, we will return ACTION_REQUIRED, because it will allow user to open the capability in the store and make changes.
If there are none that are action_required, we return pending if there are any in pending or approved, and return granted otherwise

Reply:
```
documentation: `
        Check statuses of all prerequisite capabilities - returning:
        GRANTED: If all pre-reqs are in granted status
        PENDING : At least one pre-req is still in pending/approved status
        ACTION_REQUIRED: If not any of the above
`
```
How does it sound? @noodlemoodle @KernelDeimos

Reply:
I think for PENDING documentation, there should be one more condition of `all other prereqs not in pending/approved are in granted status`, or swap pending and action_required so that : 
action_required = atleast one prereq is not in pending/approved/granted status
pending = none of the above
because if some prereqs are in pending and some are action_required, we would want to return action_required.

but everything else LGTm

Reply:
Updated in https://github.com/foam-framework/foam2/pull/4010/commits/003a198c4e70f0fa24bf28672e1bcf4fbe23e27e.